python/cuml/ensemble/randomforest_common.pyx

```diff
@@ -0,0 +1,448 @@
+        tl_model = TreeliteModel.from_treelite_model_handle(
+            self.treelite_handle,
+            take_handle_ownership=False)
+        self.n_cols = tl_model.num_features
+        self.n_estimators = tl_model.num_trees
+        return self
+
+    def _predict_model_on_gpu(self, X, algo, convert_dtype,
+                              fil_sparse_format, threshold=0.5,
+                              output_class=False, predict_proba=False):
+        out_type = self._get_output_type(X)
+        cdef ModelHandle cuml_model_ptr = NULL
+        _, n_rows, n_cols, dtype = \
+            input_to_cuml_array(X, order='F',
+                                check_cols=self.n_cols)
+
+        if dtype == np.float64 and not convert_dtype:
+            raise TypeError("GPU based predict only accepts np.float32 data. \
+                            Please set convert_dtype=True to convert the test \
+                            data to the same dtype as the data used to train, \
+                            ie. np.float32. If you would like to use test \
+                            data of dtype=np.float64 please set \
+                            predict_model='CPU' to use the CPU implementation \
+                            of predict.")
+
+        treelite_handle = self._obtain_treelite_handle()
+
+        storage_type = \
+            _check_fil_parameter_validity(depth=self.max_depth,
+                                          fil_sparse_format=fil_sparse_format,
+                                          algo=algo)
+        fil_model = ForestInference()
+        tl_to_fil_model = \
+            fil_model.load_using_treelite_handle(treelite_handle,
+                                                 output_class=output_class,
+                                                 threshold=threshold,
+                                                 algo=algo,
+                                                 storage_type=storage_type)
+
+        preds = tl_to_fil_model.predict(X, output_type=out_type,
+                                        predict_proba=predict_proba)
+        return preds
+
+    def _get_params(self, deep):
+        params = dict()
+        for key in BaseRandomForestModel.variables:
+            if key in ['handle']:
+                continue
+            var_value = getattr(self, key, None)
+            params[key] = var_value
+        return params
+
+    def _set_params(self, **params):
+        # Resetting handle as __setstate__ overwrites with handle=None
+        self.handle.__setstate__(self.n_streams)
+        self.model_pbuf_bytes = []
+
+        if not params:
+            return self
+        for key, value in params.items():
+            if key not in BaseRandomForestModel.variables:
+                raise ValueError('Invalid parameter for estimator')
+            else:
+                setattr(self, key, value)
+        return self
+
+
+def _check_fil_parameter_validity(depth, algo, fil_sparse_format):
```

Comment:
Small docstring explaining what these functions do and why they are needed will be good for future maintenance